### PR TITLE
Attempt to fix crash on macOS 11 by using Xcode 15.4 

### LINKED ISSUE #5934: Wont open, crashes on Mac 10.15.7
### Is there an existing issue for this problem?

- [X] I have searched the existing issues

### OrcaSlicer Version

OrcaSlicer V2.1.1 Official Release

### Operating System (OS)

macOS

### OS Version

Mac OS Catalina v10.15.7

### Additional system information

Model Name:	Mac Pro
  Model Identifier:	MacPro5,1
  Processor Name:	6-Core Intel Xeon
  Processor Speed:	3.46 GHz
  Number of Processors:	2
  Total Number of Cores:	6
  L2 Cache (per Core):	256 KB
  L3 Cache:	12 MB
  Hyper-Threading Technology:	Enabled
  Memory:	32 GB
  Boot ROM Version:	144.0.0.0.0
  SMC Version (system):	1.39f11
  SMC Version (processor tray):	1.39f11
  Serial Number (processor tray):	J503407RECZJB     
  Hardware UUID:	B724234B-5D12-579C-BEE7-BEEE32CB8E85

Display Card info:

NVIDIA GeForce GT 640:

  Chipset Model:	NVIDIA GeForce GT 640
  Type:	GPU
  Bus:	PCIe
  Slot:	Slot-1
  PCIe Lane Width:	x16
  VRAM (Total):	2 GB
  Vendor:	NVIDIA (0x10de)
  Device ID:	0x0fc0
  Revision ID:	0x00a2
  ROM Revision:	MacVidCards
  Metal:	Supported, feature set macOS GPUFamily1 v4
  Displays:
Cinema HD:
  Display Type:	LCD
  Resolution:	1920 x 1200 (WUXGA - Widescreen Ultra eXtended Graphics Array)
  UI Looks like:	1920 x 1200
  Framebuffer Depth:	24-Bit Color (ARGB8888)
  Display Serial Number:	2A7203GNXMN 
  Main Display:	Yes
  Mirror:	Off
  Online:	Yes
  Rotation:	Supported
  Automatically Adjust Brightness:	No
HP 25es:
  Resolution:	1920 x 1080 (1080p FHD - Full High Definition)
  UI Looks like:	1920 x 1080 @ 60 Hz
  Framebuffer Depth:	24-Bit Color (ARGB8888)
  Display Serial Number:	3CM649032Z   
  Mirror:	Off
  Online:	Yes
  Rotation:	Supported
  Automatically Adjust Brightness:	No

### Printer

NOT applicable 

### How to reproduce

Click on OrcaSlicer and just get error... OrcaSlicer Quit Unexpctedly... so don't even open

### Actual results

Process:               OrcaSlicer [24706]
Path:                  /Applications/OrcaSlicer.app/Contents/MacOS/OrcaSlicer
Identifier:            com.softfever3d.orca-slicer
Version:               2.1.1 (???)
Code Type:             X86-64 (Native)
Parent Process:        ??? [1]
Responsible:           OrcaSlicer [24706]
User ID:               501

Date/Time:             2024-06-29 19:22:48.056 -0400
OS Version:            Mac OS X 10.15.7 (19H2)
Report Version:        12
Anonymous UUID:        A98F4D78-B17C-1ABE-F309-5EFFA4FDDAEC

Sleep/Wake UUID:       F5FBE6C3-876E-4FE6-93B9-65B0F5EA4872

Time Awake Since Boot: 3300000 seconds
Time Since Wake:       2300 seconds

System Integrity Protection: disabled

Crashed Thread:        0  Dispatch queue: com.apple.main-thread

Exception Type:        EXC_BAD_ACCESS (SIGSEGV)
Exception Codes:       KERN_INVALID_ADDRESS at 0x000000000d449000
Exception Note:        EXC_CORPSE_NOTIFY

Termination Signal:    Segmentation fault: 11
Termination Reason:    Namespace SIGNAL, Code 0xb
Terminating Process:   exc handler [24706]

VM Regions Near 0xd449000:
--> 
    __TEXT                 000000010d449000-0000000111a95000 [ 70.3M] r-x/r-x SM=COW  /Applications/OrcaSlicer.app/Contents/MacOS/OrcaSlicer

Application Specific Information:
/Applications/OrcaSlicer.app/Contents/MacOS/OrcaSlicer

Thread 0 Crashed:: Dispatch queue: com.apple.main-thread
0   com.softfever3d.orca-slicer   	0x000000010d50349e __cxx_global_var_init.609 + 14
1   dyld                          	0x00000001198a41d3 ImageLoaderMachO::doModInitFunctions(ImageLoader::LinkContext const&) + 535
2   dyld                          	0x00000001198a45de ImageLoaderMachO::doInitialization(ImageLoader::LinkContext const&) + 40
3   dyld                          	0x000000011989effb ImageLoader::recursiveInitialization(ImageLoader::LinkContext const&, unsigned int, char const*, ImageLoader::InitializerTimingList&, ImageLoader::UninitedUpwards&) + 493
4   dyld                          	0x000000011989d0b4 ImageLoader::processInitializers(ImageLoader::LinkContext const&, unsigned int, ImageLoader::InitializerTimingList&, ImageLoader::UninitedUpwards&) + 188
5   dyld                          	0x000000011989d154 ImageLoader::runInitializers(ImageLoader::LinkContext const&, ImageLoader::InitializerTimingList&) + 82
6   dyld                          	0x000000011988b6a8 dyld::initializeMainExecutable() + 199
7   dyld                          	0x0000000119890bba dyld::_main(macho_header const*, unsigned long, int, char const**, char const**, char const**, unsigned long*) + 6667
8   dyld                          	0x000000011988a227 dyldbootstrap::start(dyld3::MachOLoaded const*, int, char const**, dyld3::MachOLoaded const*, unsigned long*) + 453
9   dyld                          	0x000000011988a025 _dyld_start + 37

Thread 1:
0   libsystem_pthread.dylib       	0x00007fff6b482b68 start_wqthread + 0

Thread 2:
0   libsystem_pthread.dylib       	0x00007fff6b482b68 start_wqthread + 0

Thread 0 crashed with X86 Thread State (64-bit):
  rax: 0x0000000119920268  rbx: 0x000000010d503490  rcx: 0x00007ffee27b6c08  rdx: 0x00007ffee27b6ba0
  rdi: 0x0000000000000002  rsi: 0x00007ffee27b6b88  rbp: 0x00007ffee27b2640  rsp: 0x00007ffee27b2630
   r8: 0x0000000119920330   r9: 0x0000000000000000  r10: 0x0000000000000000  r11: 0x0000000000000000
  r12: 0x0000000000000000  r13: 0x00007ffee27b2650  r14: 0x000000000d449000  r15: 0x0000000000000000
  rip: 0x000000010d50349e  rfl: 0x0000000000010202  cr2: 0x000000000d449000
  
Logical CPU:     2
Error Code:      0x00000004 (no mapping for user data read)
Trap Number:     14
....... and so on

### Expected results

that program would start

### Project file & Debug log uploads

dont work

### Checklist of files to include

- [ ] Log file
- [ ] Project file

### Anything else?

Version 2.0 worked fine

### PR DESCRIPTION
**Pull request by Noisyfox**
Fixes #5934

Source:
https://forums.developer.apple.com/forums/thread/736203

### CODE REVIEW

**Comment by Noisyfox** [comment]
> FYI: [actions/runner-images#10121](https://github.com/actions/runner-images/issues/10121)

Yeah I know, I made this PR just so that I could test if this really works.

**Comment by RockChewer3D** [comment]
@Noisyfox is there a way to make the change in layman's terms? When I tried it it kept coming back as invalid developer directory. Thanks for all your help

**Comment by Noisyfox** [comment]
> @Noisyfox is there a way to make the change in layman's terms? When I tried it it kept coming back as invalid developer directory. Thanks for all your help

Do you have a more detailed build log? And what command did you run?

**Comment by RockChewer3D** [comment]
> > @Noisyfox is there a way to make the change in layman's terms? When I tried it it kept coming back as invalid developer directory. Thanks for all your help
> 
> Do you have a more detailed build log? And what command did you run?

@Noisyfox - I followed the breadcrumbs and was able to get the "...X86_64_PR-2002.dmg" file to loaad from the following link you posted: https://github.com/SoftFever/OrcaSlicer/actions/runs/9775636085 - Thanks a million and I appreciat ethe response!

**Comment by dderbidge** [comment]
X86_64_PR_2002.dmg also worked on macOS 11.7.10. Thanks! I'm happy to receive the same hotfixes that new computers also got for now. Thanks, @Noisyfox

**Comment by fludstank** [comment]
Working well on 11.6 here - no crash on startup. 
Thank you @Noisyfox

**Comment by jorgesantos** [comment]
v2.1.1 working on Bigsur 11.17.10 intel mac, big thank you @Noisyfox

**Comment by SoftFever** [comment]
Thank you for looking into this issue!
As planned, 15.4 is now the default version.
So closing this PR.
